app/actions/remote/post.auxilary.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {chunk} from 'lodash';
+
+import {prepareModelsForChannelPosts} from '@actions/local/post';
+import {ActionType} from '@constants';
+import DatabaseManager from '@database/manager';
+
+import {fetchPostAuthors, fetchPostsForChannel} from './post';
+
+import type {Model} from '@nozbe/watermelondb';
+
+/**
+ * processChannelPostsByTeam is to fetch posts for each channel in a team, and prepare the models for the posts.
+ * Then once all the models are prepared, write them into the database all at once.
+ * Writing posts channel by channel is time consuming, and has caused the UI to lock up significantly.
+ * @param serverUrl string the server URL
+ * @param channelIds string[]
+ * @param skipAuthors boolean
+ * @param groupLabel string
+ * @param isCRTEnabled boolean
+ */
+export async function processChannelPostsByTeam(
+    serverUrl: string,
+    channelIds: string[],
+    skipAuthors = false,
+    groupLabel?: RequestGroupLabel,
+    isCRTEnabled = false,
+): Promise<void> {
+    const prepareModelsPromises: Array<Promise<Model[]>> = [];
+    const allPosts: Post[] = [];
+
+    const chunks = chunk(channelIds, 10);
+    for await (const channelIdsChunk of chunks) {
+        const channelPromises = channelIdsChunk.map((channelId) =>
+
+            // hard-coding true for skipAuthors because we want to fetch authors
+            // and upsert later to avoid unique constraint errors
+            fetchPostsForChannel(serverUrl, channelId, true, true, groupLabel),
+        );
+        const chunkResults = await Promise.allSettled(channelPromises);
```

Comment:
Promise.all() would cause the entire post fetching within the chunk to fail if even 1 post fetch returns an error.

Reply:
Sounds good, do we usually do more than 30? I'm 0/5 on the number, just trying to figure out why 30 instead of 10 or 5

Reply:
Yes, on the extreme case, 2800 channels.  But the number 30 is something that I pluck out of the thin air. It could be 20, it could be 30 or 10.

We still haven't define what channel count for "normal" users.  So if we have that count, we can use x number based on that?